modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/item_selector_repository_entry_browser.js

```diff
@@ -188,26 +188,30 @@ AUI.add(
 					_getUploadErrorMessage: function(error) {
 						var instance = this;
 
-						var errorType = error.errorType;
-
 						var message = Liferay.Language.get('an-unexpected-error-occurred-while-uploading-your-file');
 
-						if (errorType === STATUS_CODE.SC_FILE_ANTIVIRUS_EXCEPTION) {
-							message = error.message;
-						}
-						else if (errorType === STATUS_CODE.SC_FILE_EXTENSION_EXCEPTION) {
-							message = Lang.sub(Liferay.Language.get('please-enter-a-file-with-a-valid-extension-x'), [error.message]);
-						}
-						else if (errorType === STATUS_CODE.SC_FILE_NAME_EXCEPTION) {
-							message = Liferay.Language.get('please-enter-a-file-with-a-valid-file-name');
-						}
-						else if (errorType === STATUS_CODE.SC_FILE_SIZE_EXCEPTION || errorType === STATUS_CODE.SC_UPLOAD_REQUEST_CONTENT_LENGTH_EXCEPTION) {
-							message = Lang.sub(Liferay.Language.get('please-enter-a-file-with-a-valid-file-size-no-larger-than-x'), [instance.formatStorage(instance.get('maxFileSize'))]);
-						}
-						else if (errorType === STATUS_CODE.SC_UPLOAD_REQUEST_SIZE_EXCEPTION) {
-							var maxUploadRequestSize = Liferay.PropsValues.UPLOAD_SERVLET_REQUEST_IMPL_MAX_SIZE;
+						if (error && error.errorType) {
```

Comment:
Hey @natocesarrego 
Can you explain what are these changes doing? Why are they necessary?
As far as I see these changes should not be required for this fix, but maybe I'm missing something.
Thanks!

Reply:
Hi @sergiogonzalez,

During my investigation to find the root cause of this issue, I realized that the variable error was undefined, hence the execution of [error.errorType](https://github.com/sergiogonzalez/liferay-portal/pull/4139/commits/fdc9d65f9ecaf76f6683c2b9d7fd33add2809639#diff-29ffc89675b31b4ace7ecbc2350884ddR194) will throw an error and no message will be displayed to the user. I know I fixed the use case for the error related to the file extensions, but what about the other errors that could be uncovered in the backend? The file extension error treated in this pull is an example that was not covered in the backend. Therefore, this check makes sure a generic error message will be displayed to the user at least, instead of no error message. My idea is protect this frontend code from uncovered use cases in the backend.

Best.

Reply:
One more thing, @sergiogonzalez, the reasons mentioned in my previous comment to explain the need of the first commit, should be also used to explain the need of the third commit.

Best.